unittest/test_diff_actions.cpp

```diff
@@ -20,29 +20,29 @@ using namespace crocoddyl::unittest;
 void test_check_data(DifferentialActionModelTypes::Type action_type) {
   // create the model
   DifferentialActionModelFactory factory;
-  const boost::shared_ptr<crocoddyl::DifferentialActionModelAbstract>& model = factory.create(action_type);
+  boost::shared_ptr<crocoddyl::DifferentialActionModelAbstract> model = factory.create(action_type);
 
   // Run the print function
   std::ostringstream tmp;
   tmp << *model;
 
   // create the corresponding data object
-  const boost::shared_ptr<crocoddyl::DifferentialActionDataAbstract>& data = model->createData();
+  boost::shared_ptr<crocoddyl::DifferentialActionDataAbstract> data = model->createData();
 
   BOOST_CHECK(model->checkData(data));
 }
 
 void test_calc_returns_state(DifferentialActionModelTypes::Type action_type) {
   // create the model
   DifferentialActionModelFactory factory;
-  const boost::shared_ptr<crocoddyl::DifferentialActionModelAbstract>& model = factory.create(action_type);
+  boost::shared_ptr<crocoddyl::DifferentialActionModelAbstract> model = factory.create(action_type);
 
   // create the corresponding data object
-  const boost::shared_ptr<crocoddyl::DifferentialActionDataAbstract>& data = model->createData();
+  boost::shared_ptr<crocoddyl::DifferentialActionDataAbstract> data = model->createData();
 
   // Generating random state and control vectors
-  const Eigen::VectorXd& x = model->get_state()->rand();
-  const Eigen::VectorXd& u = Eigen::VectorXd::Random(model->get_nu());
+  const Eigen::VectorXd x = model->get_state()->rand();
```

Comment:
same question here?

Reply:
I don't think that rand() is returning a const-ref to something that is stored / a member variable. We may run the risk of reference to a temporary. What are your thoughts @jcarpent ?